content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,336 @@
+- `transport-message-auth`: is set to `0`.
+
+When a `transport-message` corresponding to `protocol-id = 254, 255` is retrieved, 
+it SHOULD be decoded as the `data` field in [26/WAKU-PAYLOAD](/spec/26) specification.
+
+## Appendix: Supported Handshakes Description
+
+
+Supported Noise handshakes address four typical scenarios occurring when an encrypted communication channel between Alice and Bob is going to be created: 
+
+- Alice and Bob know each others' static key.
+- Alice knows Bob's static key;
+- Alice and Bob share no key material and they don't know each others' static key.
+- Alice and Bob share some key material, but they don't know each others' static key.
+
+
+**Adversarial Model**: an active attacker who compromised one party's static key may lower the identity-hiding security guarantees provided by some handshakes. In our security model we exclude such adversary, but for completeness we report a summary of possible de-anonymization attacks that can be performed by an active attacker.
+
+### The `K1K1` Handshake
+
+If Alice and Bob know each others' static key (e.g., these are public or were already exchanged in a previous handshake) , they MAY execute a `K1K1` handshake.  Using [Noise notation](https://noiseprotocol.org/noise.html#overview-of-handshake-state-machine) *(Alice is on the left)* this can be sketched as:
+
+```
+ K1K1:
+    ->  s
+    <-  s
+       ...
+    ->  e
+    <-  e, ee, es
+    ->  se
+```
+
+We note that here only ephemeral keys are exchanged. This handshake is useful in case Alice needs to instantiate a new separate encrypted communication channel with Bob, e.g. opening multiple parallel connections, file transfers, etc.
+
+**Security considerations on identity-hiding (active attacker)**: no static key is transmitted, but an active attacker impersonating Alice can check candidates for Bob's static key.
+
+### The `XK1` Handshake
+
+Here, Alice knows how to initiate a communication with Bob and she knows his public static key: such discovery can be achieved, for example, through a publicly accessible register of users' static keys, smart contracts, or through a previous public/private advertisement of Bob's static key.
+
+A Noise handshake pattern that suits this scenario is `XK1`:
+
+```
+ XK1:
+    <-  s
+       ...
+    ->  e
+    <-  e, ee, es
+    ->  s, se
+```
+
+Within this handshake, Alice and Bob reciprocally authenticate their static keys `s` using ephemeral keys `e`. We note that while Bob's static key is assumed to be known to Alice (and hence is not transmitted), Alice's static key is sent to Bob encrypted with a key derived from both parties ephemeral keys and Bob's static key.
+
+**Security considerations on identity-hiding (active attacker)**: Alice's static key is encrypted with forward secrecy to an authenticated party. An active attacker who impersonates Alice can check candidates for Bob's static key against recorded/accepted exchanged handshake messages.
```

Comment:
The attacker should also have access to some old transcripts of Alice's handshakes with others for which the attacker does not know the other end of communication, right?

Reply:
Not really, the Attacker/Alice sends an ephemeral key (knows the DLOG), receives Bob answer and wants to test for Bob static key `s`. She then tries to encrypt her static key with the key derived from `ee` (she can since knows the DLOG of her and Bob's e is public) and a guess for the result of the `es` DH (computed using the known `e` DLOG and her guess for Bob's keys). If Bob doesn't fail the authentication on Alice/Attacker encryption, the guess for Bob's static key was correct.

Reply:
I thought the attack is against both Alice and Bob. 
So the attacker does not have to impersonate Alice, I mean there is no impersonation (the attacker does not use any knowledge about Alice for this attack), the attacker can initiate the handshake on his own

Reply:
The security consideration is with respect to the handshake and each party: can we infer parties static key by passively/actively interact with the handshake?
In this example, Alice's static key is protected by strong forward secrecy, so you need to break session keys to decrypt it (out of scope). In case of Bob, an attacker can check keys candidates.
In other attacks, the Attacker may need to _really_ impersonate Alice to guess Bob's static key, i.e. may need to actively use the DLOG of Alice's static key to succeed in the attack. However in this paragraph impersonation is meant with respect to Alice's role of handshake initiator since all protocols are stated only in terms of these two parties.

Reply:
> In other attacks, the Attacker may need to really impersonate Alice to guess Bob's static key,

Thanks for your answer. Right, in other attacks, but not the one described in the spec. I think it would be better to clarify it in the spec.

Reply:
To keep the scope of this PR limited, you may create issues regarding clarification of attack scenarios. We can later discuss them in our 1:1 meetings.

Reply:
Addressed in https://github.com/vacp2p/rfc/commit/505e4055ca5537650853d4fd33e7f8bea33650c0